src/axom/sidre/core/MFEMSidreDataCollection.cpp

```diff
@@ -1078,7 +1083,15 @@ void MFEMSidreDataCollection::Save(const std::string& filename,
     }
     temp_domain_grp->copyGroup(domain_grp);
 
-    writer.write(temp_root, num_procs, file_path, protocol);
+    int num_files = num_procs;
+    if(m_num_files > 0)
+    {
+      SLIC_ASSERT_MSG(num_files <= num_procs,
+                      "Save output must have num_files less than or equal to "
+                      "number of mpi ranks");
+      num_files = m_num_files;
```

Comment:
Since asserts are only active in debug configurations, please set `num_files` to at most `num_procs`

We have several utility functions that might be helpful, e.g. `axom::utilities::min()` or `axom::utiliities::clampVal()`
See: https://github.com/LLNL/axom/blob/57558853d1cc9193387d332faa1e4269b6589827/src/axom/core/utilities/Utilities.hpp